clients/unshaded/src/main/java/tachyon/client/ClientOptions.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.client;
+
+import tachyon.Constants;
+import tachyon.conf.TachyonConf;
+import tachyon.thrift.NetAddress;
+
+/**
+ * Represents the set of operation specific configuration options a user can pass into a Tachyon
+ * client. Not all options will be valid for all operations.
+ */
+public class ClientOptions {
+
+  /**
+   * Builder for the <code>ClientOptions<</code>.
+   */
+  public static class Builder {
+    private long mBlockSize;
+    private TachyonStorageType mTachyonStorageType;
+    private UnderStorageType mUnderStorageType;
+    private NetAddress mLocation;
+
+    /**
+     * @param conf Tachyon configuration
+     */
+    public Builder(TachyonConf conf) {
+      mBlockSize = conf.getBytes(Constants.USER_DEFAULT_BLOCK_SIZE_BYTE);
+      mTachyonStorageType =
+          conf.getEnum(Constants.USER_DEFAULT_TACHYON_STORAGE_TYPE, TachyonStorageType.STORE);
+      mUnderStorageType =
+          conf.getEnum(Constants.USER_DEFAULT_UNDER_STORAGE_TYPE, UnderStorageType.NO_PERSIST);
+      mLocation = null;
+    }
+
+    /**
+     * @param tachyonStorageType the Tachyon storage type to use
+     * @return the builder
+     */
+    public Builder setCacheType(TachyonStorageType tachyonStorageType) {
+      mTachyonStorageType = tachyonStorageType;
+      return this;
+    }
+
+    /**
+     * @param location the location to use
+     * @return the builder
+     */
+    public Builder setLocation(NetAddress location) {
+      throw new UnsupportedOperationException("Set location is currently unsupported.");
+    }
+
+    /**
+     * @param underStorageType the under storage type to use
+     * @return the builder
+     */
+    public Builder setUnderStorageType(UnderStorageType underStorageType) {
```

Comment:
Would be great to rename `setCacheType` to `setTachyonStorageType` for consistency, and have another method like `setStorageType(TachyonStorageType, UnderStorageType)` since it's more convenient and compact to specify both types of storage.